### ISSUE TITLE
add multiple fonts bug

### ISSUE THREAD

**Issue by yuhenmeili** [ocornut/imgui#6276]
version dear imgui, v1.89.2 WIP

here is my code:

<img width="1701" alt="企业微信截图_20230327005919" src="https://user-images.githubusercontent.com/8412761/227792186-ca21a363-d13f-48ed-a9dc-222e587d88be.png">

```
std::vector<std::string> files = { "../../misc/fonts/Roboto-Medium.ttf" , "../../misc/fonts/DroidSans.ttf" , "../../misc/fonts/Roboto-Medium.ttf" };
    for (int i = 0; i <= 2; ++i)
    {
        ImVector<ImWchar> ranges;
        ImFontGlyphRangesBuilder builder;

        std::string testValue = "1234567890-，锁车失败已闭锁UWB智能APP蓝牙钥匙遗留车内主副驾左右侧后备箱区域未检测到有效长刷可关闭门并锁异常需手动紧急靠边停远离辆动力电池热失控";
        std::string testValue1 = "刹屏胎缆毂匝桩℃氛";
        builder.AddText(testValue.c_str());
        builder.AddText(testValue1.c_str());
        builder.AddRanges(io.Fonts->GetGlyphRangesChineseSimplifiedCommon());
        builder.BuildRanges(&ranges);


        ImFontConfig fontConfig = ImFontConfig();
        fontConfig.OversampleH = 2;
        fontConfig.OversampleV = 1;
        fontConfig.GlyphRanges = ranges.Data;
        fontConfig.SizePixels = i == 2 ? 120.0f : 32.0f;

        io.Fonts->AddFontFromFileTTF(files[i].c_str(), 0.0f, &fontConfig, 0);
    }
    io.Fonts->Flags |= ImFontAtlasFlags_NoPowerOfTwoHeight;
    io.Fonts->Build();
```

and the problem is this font load fail
<img width="1346" alt="企业微信截图_20230327005527" src="https://user-images.githubusercontent.com/8412761/227791836-3613c9c7-7224-450a-b774-bf395266d853.png">

**Comment by emoon** [comment]
https://github.com/ocornut/imgui/blob/master/imgui.h#L2778

**Comment by yuhenmeili** [comment]
tks lot @emoon ，that's really help me. 
![image](https://user-images.githubusercontent.com/8412761/227867565-5a74a419-c7b8-4cb8-a4de-16c480b42379.png)

**Comment by ocornut** [comment]
I apologize for this old api/constraint, my main reason for it was to avoid copying ranges generated by our helper functions to retrieve large Chinese range. Next api design should lift this constraint.

**Comment by emoon** [comment]
np! I actually deleted my reply because I thought it was wrong, but now I realizied it wasn't :D anyway, I have ran into the same issue in the past and I'm happy to see that the new API will address this issue.

**Comment by ocornut** [comment]
FYI, work on `dynamic_fonts` branch aimed to be in merged 1.92 (#8465) indirectly solves this, as specifying a GlyphRange tends to be unnecessary now.
